python/cudf/cudf/core/column_accessor.py

```diff
@@ -19,12 +20,60 @@
 
 import cudf
 from cudf.core import column
-from cudf.utils.utils import cached_property, to_flat_dict, to_nested_dict
+from cudf.utils.utils import cached_property
 
 if TYPE_CHECKING:
     from cudf.core.column import ColumnBase
 
 
+class _NestedGetItemDict(dict):
+    """A dictionary whose __getitem__ method accesses nested dicts.
+
+    This class directly subclasses dict for performance, so there are a number
+    of gotchas: 1) the only safe accessor for nested elements is
+    `__getitem__` (all other accessors will fail to perform nested lookups), 2)
```

Comment:
Should we "poison" other accessor methods with something like a `RuntimeError` just to make sure someone can't inadvertently use this?

Reply:
I considered this, and I'm not sure that it's worth it. The `_grouped_data` attribute is accessed in just two places, so we'd be overriding a lot of methods, and any improper use of this data structure will cause failures almost immediately in any multi-indexed column tests. Additionally, any such change is likely to be transient since we're looking at other rewrites to the `ColumnAccessor` internals anyway. That said, I can change if you still think it's worthwhile.

Reply:
Thanks for the explanation. Lets punt on it until we have a clearer path as far as what the Column Accessor will eventually look like.